src/app/pages/PerpetualPage/components/CompetitionPage/components/Leaderboard/index.tsx

```diff
@@ -102,16 +100,10 @@ export const Leaderboard: React.FC<ILeaderboardProps> = ({
             let unrealizedProfit = 0;
 
             if (trader.positions.find(item => !item.isClosed)) {
-              const traderState = await bridgeNetwork
-                .call(
-                  Chain.BSC,
-                  contract.address,
-                  contract.abi,
-                  'getTraderState',
-                  [perpetualId, item.walletAddress.toLowerCase()],
-                )
-                .then(result => parseTraderState(result))
-                .catch(console.error);
+              const traderState = traderStates.find(
+                traderState =>
+                  traderState.id === item.walletAddress.toLowerCase(),
```

Comment:
traderState.id is not the walletAddress but an extension of it.
```suggestion
                  traderState.id.startsWith(item.walletAddress.toLowerCase()),
```